components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/publishers/WSO2APIPublisher.java

```diff
@@ -131,54 +130,34 @@ private void evaluateCredentialsAndEndpointURL(APIStore store) throws APIManagem
+            // Get tenant domain and ID to generate the original DevPortal URL (redirect URL) for the original Store
             tenantDomain = MultitenantUtils.getTenantDomain(APIUtil.replaceEmailDomainBack(
                     api.getId().getProviderName()));
             tenantId = ServiceReferenceHolder.getInstance().getRealmService().getTenantManager()
                     .getTenantId(tenantDomain);
-            api.setRedirectURL(getExternalStoreRedirectURLForAPI(tenantId, api.getUUID()));
-            //remove custom sequences from API to prevent them getting deployed in external store gateway
-            api.setInSequence(null);
-            api.setOutSequence(null);
-            api.setFaultSequence(null);
+
             //Export API as an archive file and set it as a multipart entity in the request
-            archiveBasePath = apiImportExportManager.exportAPIArtifacts(api, Boolean.TRUE, ExportFormat.JSON);
-            //remove endpoint_certificates.json if exists to prevent adding them into external store trustore
-            File endpointCertificates = new File(archiveBasePath + File.separator + api.getId().getApiName() + "-"
-                    + api.getId().getVersion() + APIImportExportConstants.JSON_ENDPOINTS_CERTIFICATE_FILE);
-            if (endpointCertificates.exists()) {
-                FileUtils.deleteQuietly(endpointCertificates);
-            }
-            CommonUtil.archiveDirectory(archiveBasePath);
-            apiArchive = new File(archiveBasePath + APIConstants.ZIP_FILE_EXTENSION);
+            ImportExportAPI importExportAPI = APIImportExportUtil.getImportExportAPI();
+            apiArchive = importExportAPI.exportAPI(api.getUuid(), api.getId().getName(), api.getId().getVersion(),
+                    String.valueOf(api.getRevisionId()), api.getId().getProviderName(), Boolean.TRUE, ExportFormat.JSON,
+                    Boolean.TRUE, Boolean.TRUE, Boolean.FALSE,
+                    getExternalStoreRedirectURLForAPI(tenantId, api.getUuid()));
             if (log.isDebugEnabled()) {
                 log.debug("API successfully exported to file: " + apiArchive.getName());
             }
         } catch (APIImportExportException e) {
             String errorMessage = "Error while exporting API: " + api.getId().getApiName() + " version: "
                     + api.getId().getVersion();
-            log.error(errorMessage, e);
             throw new APIManagementException(errorMessage, e);
         } catch (UserStoreException e) {
-            String errorMessage = "Error while getting tenantId for tenant domain: " + tenantDomain
-                    + " when exporting API:" + api.getId().getApiName() + " version: " + api.getId().getVersion();
+            String errorMessage =
+                    "Error while getting tenantId for tenant domain: " + tenantDomain + " when exporting API:" + api.getId().getApiName() + " version: " + api.getId().getVersion();
             log.error(errorMessage, e);
```

Comment:
Since we are not logging the `APIImportExportException` above, shall we also avoid logging the `UserStoreException` here?

Reply:
Fixed via https://github.com/wso2/carbon-apimgt/pull/10141/commits/0d054ff2d50f622137dfdd44e5b0ba035f8d8f26